app/lib/world/GoalManager.coffee

```diff
@@ -96,6 +105,37 @@ module.exports = class GoalManager extends CocoClass
       @goalStates[goalID] = goalState
     @notifyGoalChanges()
 
+  # Adds any goals for the current capstoneStage
+  # Returns the current capstoneStage
+  addAdditionalGoals: (session, additionalGoals) ->
+    state = session.get('state') ? {}
+    if not state.capstoneStage
+      state.capstoneStage = 1 # Increase from zero
+    else
+      state.capstoneStage += 1
+    goalsAdded = false
+    _.forEach(additionalGoals, (stageGoals) =>
+      if stageGoals.stage == state.capstoneStage
+        _.forEach(stageGoals.goals, (goal) =>
+          goalsAdded = true
```

Comment:
Could this increase the capstone level when an old goal is incomplete, then complete again?

Reply:
Sorry I dont understand this question. Also can you explain why we need `goalsAdded` flag?

Reply:
This was added as the loading of additional goals for `GoalManager`'s was more in flux - will loop back over this to verify that it is not needed 👍